wal.go

```diff
@@ -858,7 +885,18 @@ func (w *WAL) truncateHeadLocked(newMin uint64) error {
 		return fin, postCommit, nil
 	})
 
-	return w.mutateStateLocked(txn)
+	for {
+		err := w.mutateStateLocked(txn)
+		switch err {
+		case nil:
+			return nil
+		case ErrUndeletable:
```

Comment:
should there be a deadline for this ? what if it gets stuck?

Reply:
My thinking is, if the deletable stuff is enabled, it's the library user's (i.e. Vault's) job to ensure that it won't get stuck, by discovering sealed segments in a timely manner, and flagging them as deletable quickly.

Reply:
Hmm, what do you mean by quickly?

Reply:
Within a few seconds.